internal/runnerinstall/k8s.go

```diff
@@ -224,15 +237,149 @@ func (i *K8sRunnerInstaller) InstallFlags(set *flag.Set) {
 }
 
 func (i *K8sRunnerInstaller) Uninstall(ctx context.Context, opts *InstallOpts) error {
-	//TODO implement me
-	panic("implement me")
+	sg := opts.UI.StepGroup()
+	defer sg.Wait()
+
+	s := sg.Add("Preparing Helm...")
+	defer func() { s.Abort() }()
+
+	actionConfig, err := helminstallutil.ActionInit(opts.Log, i.Config.KubeconfigPath, i.Config.K8sContext)
+	if err != nil {
+		return err
+	}
+	s.Update("Helm settings retrieved")
+	s.Status(terminal.StatusOK)
+	s.Done()
+
+	s = sg.Add("Uninstallation Pre-check...")
+	helmRunnerId := "waypoint-" + strings.ToLower(opts.Id)
+	verifyClient := action.NewGetValues(actionConfig)
+	cfg, err := verifyClient.Run(helmRunnerId)
+	if err != nil {
+		return err
+	}
+
+	var runnerCfg InstalledRunnerConfig
+	err = mapstructure.Decode(cfg["runner"], &runnerCfg)
+	if err != nil {
+		return err
+	}
+
+	// Check if the runner we are uninstalling matches the helm chart
+	// This should always be true and is a sanity check to make sure this is a
+	// proper runner installation and that we are uninstalling what we think we
+	// should be uninstalling.
+	if strings.ToLower(runnerCfg.Id) != strings.ToLower(opts.Id) {
+		opts.Log.Debug("Could not find runner for id %q", opts.Id)
+		return errors.New("Runner not found")
+	}
+	s.Update("Runner %q found", opts.Id)
+	s.Status(terminal.StatusOK)
+	s.Done()
+
+	s = sg.Add("Uninstalling Runner...")
+	client := action.NewUninstall(actionConfig)
+	client.DryRun = false
+	client.DisableHooks = false
+	client.Wait = true
+	client.Timeout = 300 * time.Second
+	client.Description = ""
+
+	_, err = client.Run(helmRunnerId)
+	if err != nil {
+		return err
+	}
+	s.Update("Runner Uninstalled")
+	s.Status(terminal.StatusOK)
+	s.Done()
+
+	if ! i.Config.KeepPV {
```

Comment:
This is a great short-term solution! I think in the long term, if/when we update the Helm chart to do it, we should leverage Helm to do that for us on uninstall.